mswindows/osgeo4w/build_osgeo4w.sh

```diff
@@ -0,0 +1,193 @@
+    --with-blas \
+    --with-lapack-includes=/mingw64/include/lapack \
+    --with-freetype \
+    --with-freetype-includes=/mingw64/include/freetype2 \
+    --with-proj-share=${OSGEO4W_ROOT_MSYS}/share/proj \
+    --with-proj-includes=${OSGEO4W_ROOT_MSYS}/include \
+    --with-proj-libs=${OSGEO4W_ROOT_MSYS}/lib \
+    --with-postgres \
+    --with-postgres-includes=${OSGEO4W_ROOT_MSYS}/include \
+    --with-postgres-libs=${OSGEO4W_ROOT_MSYS}/lib \
+    --with-gdal=${SRC}/mswindows/osgeo4w/gdal-config \
+    --with-geos=${SRC}/mswindows/osgeo4w/geos-config \
+    --with-sqlite \
+    --with-sqlite-includes=${OSGEO4W_ROOT_MSYS}/include \
+    --with-sqlite-libs=${OSGEO4W_ROOT_MSYS}/lib \
+    --with-regex \
+    --with-nls \
+    --with-zstd \
+    --with-odbc \
+    --with-cairo \
+    --with-cairo-includes=${SRC}/include \
+    --with-cairo-ldflags="-L${SRC}/mswindows/osgeo4w/lib -lcairo -lfontconfig" \
+    --with-opengl=windows \
+    --with-bzlib \
+    --with-liblas=${SRC}/mswindows/osgeo4w/liblas-config \
+    --with-netcdf=${OSGEO4W_ROOT_MSYS}/bin/nc-config
+
+make
+
+# install
+
+bin=bin.${ARCH}
+dist=dist.${ARCH}
+ver=$(sed -n '/^INST_DIR[ \t]*=/{s/^.*grass//; p}' include/Make/Platform.make)
+
+rm -f $dist/grass$ver.tmp $dist/etc/fontcap
+
+if [ "$UNITTEST" ]; then
+    # Add executables for bash scripts in unittests
+    bash_bin=bash_bin
+    mkdir "${SRC}/dist.${ARCH}/$bash_bin"
+
+    for f in ${SRC}/dist.${ARCH}/scripts/*.py
+    do
+        bash_exe=$(echo $f | sed -e "s/\/scripts\//\/${bash_bin}\//;s/\.py$//")
+        cp "$f" "$bash_exe"
+        # dos2unix "$bash_exe"
+        chmod ugo+x "$bash_exe"
+    done
+    bash_exe_path=$(echo "${SRC}/dist.${ARCH}/$bash_bin" | sed -e "s/^\/c/\;c:/;s/^\/d/d:/")
+    export PATH="$PATH:${SRC}/dist.${ARCH}/$bash_bin"
+fi
+
+# create batch files
+src_esc=$(echo ${SRC} | sed 's#^/\([a-z]\)#\1:#; s#/#\\\\\\\\#g')
+dist_esc="$src_esc\\\\$dist"
+
+(
+    sed 's/^\(set GISBASE=\).*/\1'$dist_esc'/' \
+        mswindows/osgeo4w/env.bat.tmpl
+    cat <<EOT
+
+
+set PATH=%PATH%;C:\\msys64\\mingw64\\bin;C:\\msys64\\usr\\bin
+
+if not exist %GISBASE%\etc\fontcap (
+	pushd .
+	%~d0
+	cd %GISBASE%\lib
+	set GISRC=dummy
+	%GISBASE%\bin\g.mkfontcap.exe
+	popd
+)
+EOT
+) >$dist/etc/env.bat
+unix2dos $dist/etc/env.bat
+
+(
+    sed -e 's/^\(call "\)%~dp0\(.*\)$/\1C:\\OSGeo4W\\bin\2/' \
+        -e 's/^\(call "\).*\(\\etc\\env\.bat"\)$/\1'$dist_esc'\2/' \
+        -e 's/^\(.* "\)%GISBASE%\\etc\(\\grass.*\)$/\1%GISBASE%\\..\\'$bin'\2/' \
+        -e 's/@POSTFIX@/'$ver'/g' \
+        mswindows/osgeo4w/grass.bat.tmpl
+) >$bin/grass$ver.bat
+unix2dos $bin/grass$ver.bat
+
+opt_path=${OSGEO4W_ROOT_MSYS}/opt
+grass_path=$opt_path/grass
+
+if [ "$UNITTEST" ]; then
+    msys_path=";C:/msys64/usr/bin;C:/msys64/mingw64/bin"
+fi
+
+mkdir -p $opt_path
+cp -a $dist $grass_path
+# have a versionless and versioned startup script
+cp -a $bin/grass.py $bin/grass$ver.py
+cp -a $bin/grass$ver.py $grass_path/etc
```

Comment:
Do we still keep `$ver` in this case? In fact, I tried to rename it to simply `$grass_path/etc/grass.py` and when it was called from `$grass_path/grass.bat`, it was not able to import `grass.grassdb.checks` saying `grass` is not a package. I think that's because of `grass.py` trying to import `grass.*` from `$grass_path/etc/python/grass/*` when its name itself is `grass` (importing itself instead of `$grass_path/etc/python/grass/*`).

Reply:
Yes, this is a problem. We need to resolve it some point.